UM/FileProvider.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2020 Ultimaker B.V.
+# Uranium is released under the terms of the LGPLv3 or higher.
+from PyQt5.QtCore import pyqtSignal
+
+from UM.PluginObject import PluginObject
+from typing import Optional
+
+
+class FileProvider(PluginObject):
+    """Base class for plugins that aim to provide a file to Cura in an alternate fashion, other than using the local file
+    explorer.
+
+    Every new file provider adds an option to the Open File(s) menu.
+    """
+
+    enabledChanged = pyqtSignal()
+    """Signal which informs whether the file provider has been enabled or disabled, so that it can be removed or added
+    in the Open File(s) submenu"""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._menu_item_display_text = None  # type: Optional[str]
+        self._menu_item_name = None  # type: Optional[str]
+        self._shortcut = None  # type: Optional[str]
+        self._enabled = True
+
+    def setMenuItemDisplayText(self, display_text: str) -> None:
```

Comment:
Rather than using setters and getters, which are public and not intended to be overridden by subclasses, it is simpler to just have a public field. That's what they're for, after all.

Reply:
Fixed here:
* https://github.com/Ultimaker/Uranium/pull/660/commits/ce0d2728cb04f987078ffb0f5d5d0ffbef1373a7
* https://github.com/Ultimaker/cura-private-data/pull/66/commits/46902a45a7173b5e857736c2943147eb561dc123